File: benchmarks/node/primes.js
const LIMIT = 100000;
const sieve = new Array(LIMIT + 1).fill(true);
sieve[0] = false;
sieve[1] = false;

for (let p = 2; p * p <= LIMIT; p++) {
  if (sieve[p]) {
    for (let m = p * p; m <= LIMIT; m += p) {
      sieve[m] = false;
    }
  }
}

let count = 0;
for (let k = 2; k <= LIMIT; k++) {
  if (sieve[k]) count++;
}

console.log(count);

File: benchmarks/python/primes.py
LIMIT = 100000

sieve = [True] * (LIMIT + 1)
sieve[0] = False
sieve[1] = False

p = 2
while p * p <= LIMIT:
    if sieve[p]:
        for m in range(p * p, LIMIT + 1, p):
            sieve[m] = False
    p += 1

count = 0
for k in range(2, LIMIT + 1):
    if sieve[k]:
        count += 1

print(count)
